pkg/internal/loader/disk/disk_test.go

```diff
@@ -195,13 +195,136 @@ metadata:
+				loader := NewComponents(tmp)
 				components, err := loader.Load(context.Background())
-
 				assert.Equal(t, test.expErr, err != nil, "%v", err)
 				assert.Equal(t, test.expComps, components)
 			})
 		}
 	})
 }
+
+func Test_loadWithOrder(t *testing.T) {
+	t.Run("no file should return empty set", func(t *testing.T) {
+		tmp := t.TempDir()
+		d := NewComponents(tmp).(*disk[compapi.Component])
+		set, err := d.loadWithOrder()
+		require.NoError(t, err)
+		assert.Empty(t, set.order)
+		assert.Empty(t, set.ts)
+	})
+
+	t.Run("single manifest file should return", func(t *testing.T) {
+		tmp := t.TempDir()
+		require.NoError(t, os.WriteFile(filepath.Join(tmp, "test-component.yaml"), []byte(`
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: statestore
+spec:
+  type: state.couchbase
+`), fs.FileMode(0o600)))
+
+		d := NewComponents(tmp).(*disk[compapi.Component])
+		set, err := d.loadWithOrder()
+		require.NoError(t, err)
+		assert.Equal(t, []manifestOrder{
+			{dirIndex: 0, fileIndex: 0, manifestIndex: 0},
+		}, set.order)
+		assert.Len(t, set.ts, 1)
+	})
+
+	t.Run("3 manifest file should have order set", func(t *testing.T) {
+		tmp := t.TempDir()
+		require.NoError(t, os.WriteFile(filepath.Join(tmp, "test-component.yaml"), []byte(`
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: statestore
+spec:
+  type: state.couchbase
+---
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: statestore
+spec:
+  type: state.couchbase
+---
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: statestore
+spec:
+  type: state.couchbase
+`), fs.FileMode(0o600)))
+
+		d := NewComponents(tmp).(*disk[compapi.Component])
+		set, err := d.loadWithOrder()
+		require.NoError(t, err)
+		assert.Equal(t, []manifestOrder{
+			{dirIndex: 0, fileIndex: 0, manifestIndex: 0},
+			{dirIndex: 0, fileIndex: 0, manifestIndex: 1},
+			{dirIndex: 0, fileIndex: 0, manifestIndex: 2},
+		}, set.order)
+		assert.Len(t, set.ts, 3)
+	})
+
+	t.Run("3 dirs, 3 files, 3 manifests should return order. Skips manifests of different type", func(t *testing.T) {
+		tmp1, tmp2, tmp3 := t.TempDir(), t.TempDir(), t.TempDir()
+
+		for _, dir := range []string{tmp1, tmp2, tmp3} {
+			for _, file := range []string{"1.yaml", "2.yaml", "3.yaml"} {
+				require.NoError(t, os.WriteFile(filepath.Join(dir, file), []byte(`
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: statestore
+spec:
+  type: state.couchbase
+---
+apiVersion: dapr.io/v1alpha1
+kind: Subscription
+metadata:
+  name: statestore
+spec:
+  type: state.couchbase
+---
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: statestore
+spec:
+  type: state.couchbase
+`), fs.FileMode(0o600)))
+			}
+		}
+
+		d := NewComponents(tmp1, tmp2, tmp3).(*disk[compapi.Component])
+		set, err := d.loadWithOrder()
+		require.NoError(t, err)
+		assert.Equal(t, []manifestOrder{
+			{dirIndex: 0, fileIndex: 0, manifestIndex: 0},
```

Comment:
Can you explain why the `manifestIndex` goes from 0 -> 2 here?

Reply:
Indeed- we have 3 directories, writing 3 files each, writing 3 manifests each- so index 0,1,2 on each. Since we are creating a `Component` disk loader, the test is ensuring that the `Subscription` manifest at index `1` is ignored. Hence why `manifestIndex` `1` is skipped and not expected on the result.